From 0bf5479ff7f2ce62472e2c56f66b4e43057defb4 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Thu, 8 Sep 2022 10:34:35 +0300
Subject: [PATCH] lx2160acex7: s1-21 s2-13: add dpmac17

dpmac17 and dpmac18 can be used for RGMII / SGMII independent from the
serdes, on all serdes configurations not utilizing them.
Enable dpmac17, and add comments for dpmac18.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 .../CEX7/dpc-S1_21-S2_13-6x_25gbe-2x_sgmii.dts   | 15 +++++++++++++++
 .../CEX7/dpl-S1_21-S2_13-6x_25gbe-2x_sgmii.dts   | 16 ++++++++++++++--
 2 files changed, 29 insertions(+), 2 deletions(-)

diff --git a/config/lx2160a/CEX7/dpc-S1_21-S2_13-6x_25gbe-2x_sgmii.dts b/config/lx2160a/CEX7/dpc-S1_21-S2_13-6x_25gbe-2x_sgmii.dts
index 2134252..16e1d1a 100644
--- a/config/lx2160a/CEX7/dpc-S1_21-S2_13-6x_25gbe-2x_sgmii.dts
+++ b/config/lx2160a/CEX7/dpc-S1_21-S2_13-6x_25gbe-2x_sgmii.dts
@@ -71,6 +71,7 @@
 
 	board_info {
 		ports {
+			/* Serdes 1 */
 			mac@3 {
 				link_type = "MAC_LINK_TYPE_FIXED";
 			};
@@ -89,12 +90,26 @@
 			mac@10 {
 				link_type = "MAC_LINK_TYPE_FIXED";
 			};
+
+			/* Serdes 2 */
 			mac@13 {
 				link_type = "MAC_LINK_TYPE_PHY";
 			};
 			mac@14 {
 				link_type = "MAC_LINK_TYPE_PHY";
 			};
+
+			/* EC1 */
+			mac@17 {
+				/* dpmac17 can be used for sgmii/rgmii, independent from (some) serdes configuration(s) */
+				link_type = "MAC_LINK_TYPE_PHY";
+			};
+
+			/* EC2 */
+			/* mac@18 {
+				/* dpmac18 can be used for sgmii/rgmii, independent from (some) serdes configuration(s) */
+				link_type = "MAC_LINK_TYPE_PHY";
+			};*/
 		};
 	};
 };
diff --git a/config/lx2160a/CEX7/dpl-S1_21-S2_13-6x_25gbe-2x_sgmii.dts b/config/lx2160a/CEX7/dpl-S1_21-S2_13-6x_25gbe-2x_sgmii.dts
index 8351128..886eb7c 100644
--- a/config/lx2160a/CEX7/dpl-S1_21-S2_13-6x_25gbe-2x_sgmii.dts
+++ b/config/lx2160a/CEX7/dpl-S1_21-S2_13-6x_25gbe-2x_sgmii.dts
@@ -39,7 +39,7 @@
 			objects {
 				obj_set@dpmac {
 					type = "dpmac";
-					ids = <0x3 0x4 0x5 0x6 0x9 0xa 0xd 0xe>;
+					ids = <0x3 0x4 0x5 0x6 0x9 0xa 0xd 0xe 0x11>;
 				};
 				obj_set@dpmcp {
 					type = "dpmcp";
@@ -53,7 +53,7 @@
 	 * Objects
 	 *****************************************************************/
 	objects {
-
+		/* Serdes 1 */
 		dpmac@3 {
 		};
 		dpmac@4 {
@@ -66,11 +66,23 @@
 		};
 		dpmac@10 {
 		};
+
+		/* Serdes 2 */
 		dpmac@13 {
 		};
 		dpmac@14 {
 		};
 
+		/* EC1 */
+		/* dpmac17 can be used for sgmii/rgmii, independent from (some) serdes configuration(s) */
+		dpmac@17 {
+		};
+
+		/* EC2 */
+		/* dpmac18 can be used for sgmii/rgmii, independent from (some) serdes configuration(s) */
+		/*dpmac@18 {
+		};*/
+
 		dpmcp@1 {
 		};
 		dpmcp@2 {
-- 
2.37.3

